frontend/src/components/debiai/dataAnalysis/algoProviders/UseAlgorithmInput.vue

```diff
@@ -210,6 +253,25 @@ export default {
       }
       this.$emit("inputValueUpdate", this.value);
     },
+    idListSelected(type) {
+      let idColumnsIndex = null;
+      for (let i = 0; i < this.data.columns.length; i++) {
+        if (this.data.columns[i].label === "Data ID") {
+          idColumnsIndex = i;
+          break;
+        }
+      }
```

Comment:
1. Do once in the `mounted` function if `isIdList` is true, store the `idColumnsIndex` as `this.idColumnsIndex` (no need to add it to the data).

2. Use `Array.prototype.findIndex()` ([more info](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex))